File: app/src/main/java/com/mancode/financetracker/ui/balances/BalanceFragment.java
package com.mancode.financetracker.ui.balances;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.mancode.financetracker.R;
import com.mancode.financetracker.database.viewmodel.BalanceViewModel;
import com.mancode.financetracker.ui.UIUtils;

/**
 * Created by Manveru on 03.09.2017.
 */

public class BalanceFragment extends Fragment {

    private BalanceRecyclerViewAdapter mAdapter;

    public BalanceFragment() { }

    public static BalanceFragment newInstance() {
        return new BalanceFragment();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        BalanceViewModel balanceViewModel =
                ViewModelProviders.of(getActivity()).get(BalanceViewModel.class);
        balanceViewModel.getNetValues().observe(this,
                netValues -> mAdapter.submitList(netValues));
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_account_list, container, false); // TODO layout to change?
        View recyclerView = view.findViewById(R.id.list);

        // Set the adapter
        if (recyclerView instanceof RecyclerView) {
            Context context = view.getContext();
            RecyclerView rView = (RecyclerView) recyclerView;
            rView.setLayoutManager(new LinearLayoutManager(context));
            mAdapter = new BalanceRecyclerViewAdapter();
            rView.setAdapter(mAdapter);
        }

        FloatingActionButton fab = view.findViewById(R.id.fab);
        fab.setOnClickListener(view1 -> UIUtils.showFullScreenDialog(
                getFragmentManager(), AddBalanceFragment.newInstance()));
        return view;
    }
}

File: app/src/main/java/com/mancode/financetracker/ui/balances/BalanceFragment.kt
package com.mancode.financetracker.ui.balances

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.NavController
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.mancode.financetracker.R
import com.mancode.financetracker.databinding.FragmentBalanceListBinding
import com.mancode.financetracker.viewmodel.BalanceViewModel
import org.threeten.bp.LocalDate

/**
 * Created by Manveru on 03.09.2017.
 */

class BalanceFragment : Fragment(R.layout.fragment_balance_list), BalanceRecyclerViewAdapter.ModifyRequestListener {

    private var _binding: FragmentBalanceListBinding? = null
    private val binding get() = _binding!!

    private lateinit var navController: NavController

    private val adapter: BalanceRecyclerViewAdapter by lazy { BalanceRecyclerViewAdapter(this) }
    private val viewModel: BalanceViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        _binding = FragmentBalanceListBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel.netValues.observe(viewLifecycleOwner, { netValues ->
            adapter.submitList(netValues)
            if (netValues.isNotEmpty()) {
                binding.balancesList.visibility = View.VISIBLE
                binding.emptyListInfo.visibility = View.GONE
            } else {
                binding.balancesList.visibility = View.GONE
                binding.emptyListInfo.visibility = View.VISIBLE
            }
        })
        navController = findNavController()
        binding.balancesList.layoutManager = LinearLayoutManager(context)
        binding.balancesList.adapter = adapter
        binding.fab.setOnClickListener {
            navController.navigate(R.id.action_balanceFragment_to_addBalanceFragment)
        }
    }

    override fun onDestroyView() {
        _binding = null
        super.onDestroyView()
    }

    override fun onEditRequested(date: LocalDate) {
        val action = BalanceFragmentDirections.actionBalanceFragmentToAddBalanceFragment(date)
        navController.navigate(action)
    }
}
